src/components/static-site.ts

```diff
@@ -36,18 +36,28 @@ export class StaticSite extends pulumi.ComponentResource {
 
     this.name = name;
     const { domain, hostedZoneId, tags } = args;
-    this.certificate = this.createTlsCertificate({ domain, hostedZoneId });
+    const hasCustomDomain = domain && hostedZoneId;
+    if (domain && !hostedZoneId) {
+      throw new Error(
+        'StaticSite:hostedZoneId must be provided when the domain is specified',
+      );
+    }
+    if (hasCustomDomain) {
+      this.certificate = this.createTlsCertificate({ domain, hostedZoneId });
+    }
     this.bucket = this.createPublicBucket({ tags });
     this.cloudfront = this.createCloudfrontDistribution({ domain, tags });
-    this.createDnsRecord({ domain, hostedZoneId });
+    if (hasCustomDomain) {
```

Comment:
Why is this not grouped with creating Tls certificate in line 46?

Reply:
It depends on Cloudfront distribution and therefore it must be called after CloudFront creation